### PR TITLE
[Mikhail-Ilichev] Tasks 21-30

### PR DESCRIPTION
**Pull request by miffn3**

### CODE REVIEW

File: tasks/22_remove-falsy-values/remove-falsy-values.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export function removeFalsyValues(arr) {
+  const result = arr.filter((word) => Boolean(word));
```
Comment (vovcyan) [inline tasks/22_remove-falsy-values/remove-falsy-values.js]: return arr.filter((word) => Boolean(word));

File: tasks/24_index-to-insert/index-to-insert.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+export function getIndexToInsert(arr, num) {
+  const result = arr.slice();
+  result.push(num);
+  result.sort((a, b) => a-b);
```
Comment (vovcyan) [inline tasks/24_index-to-insert/index-to-insert.js]: a - b, missed space

File: tasks/25_rot13-encoder/rot13-encoder.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+export function rot13Encoder(str) {
+  const encodeMessage = [];
+  const regex = /[A-Z]/;
+  for (let i = 0; i < str.length; i++) {
+    if (str[i].match(regex)) {
+      encodeMessage.push(String.fromCharCode(str.charCodeAt(i) % 26 + 65));
```
Comment (vovcyan) [inline tasks/25_rot13-encoder/rot13-encoder.js]: split it into 2 lines

File: tasks/27_basic-subclasses/basic-subclasses.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+export class God {
+  static create() {
+    return [new Man, new Woman];
```
Comment (vovcyan) [inline tasks/27_basic-subclasses/basic-subclasses.js]: Incorrect expression

**Review by vovcyan — CHANGES_REQUESTED** [review]

File: tasks/22_remove-falsy-values/remove-falsy-values.js (hunk 1 above)
Comment (KalachevDev) [reply tasks/22_remove-falsy-values/remove-falsy-values.js]: @vovcyan moreover, `return arr.filter(Boolean)`

File: tasks/25_rot13-encoder/rot13-encoder.js (hunk 3 above)
Comment (KalachevDev) [inline tasks/25_rot13-encoder/rot13-encoder.js]: Let's describe these magic number somehow by introducing const

File: tasks/29_remove-duplicates/remove-duplicates.js [hunk 5]
```diff
@@ -0,0 +1,4 @@
+export function removeDuplicates(arr) {
+  const result = arr.slice();
+  return result.filter((item, index) => result.indexOf(item) === index);
```
Comment (KalachevDev) [inline tasks/29_remove-duplicates/remove-duplicates.js]: Seem you need to use lastIndexOf here

File: tasks/21_check-letters/check-letters.js [hunk 6]
```diff
@@ -0,0 +1,10 @@
+export function checkLetters(arr) {
+  const firstWord = arr[0].toLowerCase().split('');
+  const secondWord = arr[1].toLowerCase().split('');
```
Comment (KalachevDev) [inline tasks/21_check-letters/check-letters.js]: Do you really need to call split here?

File: tasks/25_rot13-encoder/rot13-encoder.js (hunk 3 above)
Comment (KalachevDev) [inline tasks/25_rot13-encoder/rot13-encoder.js]: Also, don't hesitate to introduce other functions here for more readability. If you think that other function perfectly suits here and helps you to follow single responsibility principle - make it.

File: tasks/27_basic-subclasses/basic-subclasses.js [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+export class Human {
+  constructor(sex, name) {
+    this.sex = sex;
+    this.name = name;
+  }
+}
+
+export class Man extends Human {
+  constructor() {
+    super('male', 'Adam');
```
Comment (KalachevDev) [inline tasks/27_basic-subclasses/basic-subclasses.js]: This looks like an incorrect usage of inheritance. Are you sure every man should be Adam?
It's better to pass name from outside rather then hardcode it and put it into class definition

Comment (KalachevDev) [reply tasks/27_basic-subclasses/basic-subclasses.js]: Also, it makes sense to introduce some const variable to describe all genders (aka sex)

**Review by KalachevDev — CHANGES_REQUESTED** [review]

File: tasks/21_check-letters/check-letters.js (hunk 6 above)
Comment (alexkond) [reply tasks/21_check-letters/check-letters.js]: Not really.

File: tasks/22_remove-falsy-values/remove-falsy-values.js (hunk 1 above)
Comment (miffn3) [reply tasks/22_remove-falsy-values/remove-falsy-values.js]: changed it to  `return arr.filter(Boolean)`.

File: tasks/24_index-to-insert/index-to-insert.js (hunk 2 above)
Comment (miffn3) [reply tasks/24_index-to-insert/index-to-insert.js]: added space between

File: tasks/25_rot13-encoder/rot13-encoder.js (hunk 3 above)
Comment (miffn3) [reply tasks/25_rot13-encoder/rot13-encoder.js]: moved it to new const, that contains operations to get new letter by offset

Comment (miffn3) [reply tasks/25_rot13-encoder/rot13-encoder.js]: added new const that contains operations to get new letter by offset

File: tasks/21_check-letters/check-letters.js (hunk 6 above)
Comment (miffn3) [reply tasks/21_check-letters/check-letters.js]: Yeap, my bad, deleted both `split('')`

File: tasks/25_rot13-encoder/rot13-encoder.js (hunk 3 above)
Comment (miffn3) [reply tasks/25_rot13-encoder/rot13-encoder.js]: Got it. I'll make it in next tasks

File: tasks/27_basic-subclasses/basic-subclasses.js (hunk 4 above)
Comment (miffn3) [reply tasks/27_basic-subclasses/basic-subclasses.js]: changed, added arguments to constructor

File: tasks/27_basic-subclasses/basic-subclasses.js (hunk 7 above)
Comment (miffn3) [reply tasks/27_basic-subclasses/basic-subclasses.js]: @KalachevDev, I've added const obj for genders (but it contains only 2 of them)

File: tasks/24_index-to-insert/index-to-insert.js [hunk 8]
```diff
@@ -0,0 +1,6 @@
+export function getIndexToInsert(arr, num) {
+  const result = arr.slice();
```
Comment (vadeneev) [inline tasks/24_index-to-insert/index-to-insert.js:2]: You could do it in more shoer way like [...arr, num]

File: tasks/29_remove-duplicates/remove-duplicates.js [hunk 9]
```diff
@@ -0,0 +1,4 @@
+export function removeDuplicates(arr) {
+  const result = arr.slice();
+  return result.filter((item, index) => result.lastIndexOf(item) === index);
```
Comment (vadeneev) [inline tasks/29_remove-duplicates/remove-duplicates.js:3]: It could be implemented in faster way

**Review by vadeneev — APPROVED** [review]

File: tasks/24_index-to-insert/index-to-insert.js (hunk 8 above)
Comment (miffn3) [reply tasks/24_index-to-insert/index-to-insert.js:2]: i tried it, it's really better to read, thanks

**Review by vovcyan — APPROVED** [review]

**Review by KalachevDev — APPROVED** [review]

**Pull request merged**